### PR TITLE
Do not install PSP resources by default

### LINKED ISSUE #6549: Do not install PSP resources by default
PodSecurityPolicy is deprecated as of k8s v1.21 and will be unavailable starting k8s v1.25. We're seeing such warnings when installing linkerd under newer k8s versions:
```console
Warning: policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
podsecuritypolicy.policy/linkerd-linkerd-control-plane created
```
We have one PSP: `linkerd-linkerd-control-plane` and its corresponding Role and RoleBinding declared [here](https://github.com/linkerd/linkerd2/blob/main/charts/linkerd2/templates/psp.yaml).
The linkerd-viz extension also has bindings [here](https://github.com/linkerd/linkerd2/blob/main/viz/charts/linkerd-viz/templates/psp.yaml), and so does the multicluster extension [here](https://github.com/linkerd/linkerd2/blob/main/multicluster/charts/linkerd-multicluster/templates/psp.yaml) and the linkerd-jaeger extension [here](https://github.com/linkerd/linkerd2/blob/main/jaeger/charts/linkerd-jaeger/templates/psp.yaml).

So the idea is to wrap all these resources into a condition over a new Helm config `EnablePSP`, whose value should be `false` by default. Then people still relying on PSP can just manually enable it via a Helm or via a `--set` flag in the CLI.

Note that the [ServiceAccounts](https://github.com/linkerd/linkerd2/blob/main/test/integration/serviceaccounts/serviceaccounts_test.go) integration test will also need to be updated accordingly.

### PR DESCRIPTION
**Pull request by alpeb**
Fixes #6549

PodSecurityPolicy is deprecated as of k8s v1.21 and will be unavailable starting k8s v1.25. This was causing warnings to be displayed in `linkerd install/upgrade` and `linkerd check`.

By default, do not include the linkerd PSP resource along with its Role and RoleBinding. If the user wants that, they can by setting `enablePSP: true`, a new config introduced for this purpose.

This was done in the linkerd, linkerd-cni, linkerd-viz, multicluster and jaeger charts.

The associated checks were also removed, including the NET_ADMIN+NET_RAW capabilities check, which rely on the PSP API.

### CODE REVIEW

File: viz/cmd/testdata/install_default.golden [hunk 1]
```diff
@@ -391,52 +388,7 @@ metadata:
...
-- kind: ServiceAccount
-  name: prometheus
-  namespace: linkerd-viz
-- kind: ServiceAccount
-  name: metrics-api
-  namespace: linkerd-viz
-- kind: ServiceAccount
-  name: tap-injector
-  namespace: linkerd-viz
+
```
Comment (dadjeibaah) [inline viz/cmd/testdata/install_default.golden:391]: Tiny nit: It looks like the templating adds a new line here. This doesn't seem to affect this being applied to a cluster but I thought I should call it out.

**Review by dadjeibaah — APPROVED** [review]
This looks great! Thanks for fixing this. I left one nit but otherwise this should be good.

File: viz/cmd/testdata/install_default.golden (hunk 1 above)
Comment (alpeb) [reply viz/cmd/testdata/install_default.golden:391]: Nice catch :+1:

**Review by kleimkuhler — APPROVED** [review]
👍

**Pull request merged**